### PR TITLE
New version of PR 6788 (JText::_() Optimizations)

### PR DESCRIPTION
**Pull request by orware**
Pull Request for Issue # .
#### Testing Instructions

Hopefully this is the fastest way to create a new PR with just the changes needed for the JText::_() optimizations from #6788.

### CODE REVIEW

File: libraries/joomla/language/language.php [hunk 1]
```diff
@@ -368,8 +368,11 @@ public function _($string, $jsSafe = false, $interpretBackSlashes = true)
 		}
 		elseif ($interpretBackSlashes)
 		{
-			// Interpret \n and \t characters
-			$string = str_replace(array('\\\\', '\t', '\n'), array("\\", "\t", "\n"), $string);
+			if (strpos($string, '\\') !== false)
```
Comment (andrepereiradasilva) [inline libraries/joomla/language/language.php]: hum you are replaceing 3 different string but only searching the position of one?

**Comment by orware** [comment]
Hi Andre,

Yes that is correct...I'm only checking for the presence for at least one backslash character (which would show up in all 3 of the cases being checked against for the str_replace call). 

If there is at least one backslash then we need to process the str_replace call because there's a strong chance it'll be needed.

If there isn't at least one backslash then we don't need to process the str_replace call.

I believe that covers things correctly.

Mainly what I observed was that the vast majority of our language strings do not contain any of those 3 characters so running the str_replace call all of the time on every since language string was unnecessary (and there are lots of these calls). The extra if statement helps to avoid them with the faster strpos check.

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on c96e2a86c4be030e72936c5058075e44c80c084b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11235">issues.joomla.org/joomla-cms/11235</a>.</sub>

**Comment by bertmert** [comment]
I have tested this item :white_check_mark: successfully on c96e2a86c4be030e72936c5058075e44c80c084b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11235">issues.joomla.org/joomla-cms/11235</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11235">issues.joomla.org/joomla-cms/11235</a>.</sub>

**Pull request merged**